cupy/random/generator.py

```diff
@@ -272,14 +272,26 @@ def choice(self, a, size=None, replace=True, p=None):
             if not numpy.allclose(p_sum, 1):
                 raise ValueError('probabilities do not sum to 1')
 
-        if not replace:
-            raise NotImplementedError
-
         if size is None:
             raise NotImplementedError
         shape = size
         size = numpy.prod(shape)
 
+        if not replace and p is None:
+            if a_size < size:
+                raise ValueError(
+                    'Cannot take a larger sample than population when '
+                    '\'replace=False\'')
```

Comment:
How about using `"`?

Reply:
I used single quotes for consistency. Is that fix important?

Reply:
OK. Not important.